main.rb

```diff
@@ -1,243 +1,7 @@
-    puts 'EMPTY' if @books.empty?
-    puts(@books.map { |book| "Title: #{book.title}, Author: #{book.author}" })
-    print MAIN_PRESSENTER_MESSAGE
-    gets
-    puts
-  end
-
-  def show_all_persons
-    puts 'List of persons:'
-    puts 'EMPTY' if @persons.empty?
-    puts(@persons.map { |person| "[#{person.class}] - Name: #{person.name}, Age: #{person.age}, ID: #{person.id}" })
-    print MAIN_PRESSENTER_MESSAGE
-    gets
-    puts
-  end
-
-  def create_person
-    puts 'Do you want to create a student (1), or a teacher (2)? [Press or Input the number]:'
-    print MAIN_INPUT_MESSAGE
-    case gets.chomp.to_i
-    when 1
-      create_student
-    when 2
-      create_teacher
-    else
-      puts
-      puts 'Invalid option choosed, please choose one that is listed.'
-      puts
-      create_person
-    end
-    puts
-  end
-
-  def create_book
-    puts 'Please, enter book information below:'
-    print 'Title --> '
-    title = gets.chomp
-    print 'Author --> '
-    author = gets.chomp
-    @books << Book.new(title, author)
-    puts
-    puts 'Book created successfully!'
-    puts
-  end
-
-  def create_rental
-    puts 'Select a book from the following list:'
-    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
-    print MAIN_INPUT_MESSAGE
-    book_index = gets.chomp.to_i
-    puts
-    puts 'Select a person form the following list:'
-    @persons.each_with_index do |person, index|
-      puts "#{index}) Name: #{person.name}, Age: #{person.age}, ID: #{person.id}"
-    end
-    print MAIN_INPUT_MESSAGE
-    person_index = gets.chomp.to_i
-    puts
-    print 'Enter date of retrieval --> '
-    date = gets.chomp
-    Rental.new(date, @books[book_index], @persons[person_index])
-    puts 'Rental created successfully!'
-    puts
-  end
-
-  def show_all_rentals
-    print 'Enter the person\'s ID --> '
-    person_id = gets.chomp.to_i
-    renter = @persons.select { |person| person.id == person_id }
-    puts
-    puts "Rentals of #{renter.first.name}:"
-    puts 'EMPTY' if renter.first.rentals.empty?
-    puts(renter.first.rentals.map { |rental| "Book: #{rental.book.title}, Rented on: #{rental.date}" })
-    print MAIN_PRESSENTER_MESSAGE
-    gets
-    puts
-  end
-
-  private
-
-  def create_student
-    print 'Age --> '
-    age = gets.chomp.to_i
-    print 'Name --> '
-    name = gets.chomp
-    print 'Has parent permission? [Y/N] --> '
-    permission = gets.chomp.upcase
-    permission = permission != 'N'
-    @persons << Student.new(age, name, permission)
-    puts 'Student created successfully!'
-    puts
-  end
-
-  def create_teacher
-    print 'Age --> '
-    age = gets.chomp.to_i
-    print 'Name --> '
-    name = gets.chomp
-    print 'Specialty --> '
-    specialty = gets.chomp
-    @persons << Teacher.new(age, name, specialty)
-    puts
-    puts 'Teacher created successfully!'
-    puts
-  end
+def main
+  App.new.main
 end
 
-App.new.main
+main
```

Comment:
- [ ] Finally, I got an error when I try to start the app. It will be nice if you could please look into it?

See screenshot.
![a1](https://user-images.githubusercontent.com/39523312/157126074-0284622a-e4c0-4292-b950-3c712585c9a7.png)

Reply:
Hello, @mmsesay, thanks for your useful and kindly feedback!, This problem is solved in commit b9be63b.
Read the Readme file and follow the instructions that are located in there.

Reply:
The pleasure is all mine @TheLuisAbarca